### PR TITLE
Update random generators

### PR DESCRIPTION
**Pull request by AaronGhost**
Update random generators to support long generation and bounded
integers.

### CODE REVIEW

File: common/src/main/java/com/graphicsfuzz/common/util/IRandom.java [hunk 1]
```diff
@@ -24,12 +24,19 @@
...
+
+  long nextLong(long bound);
+
   Float nextFloat();
 
   default int nextPositiveInt(int bound) {
     return nextInt(bound - 1) + 1;
   }
 
+
```
Comment (afd) [inline common/src/main/java/com/graphicsfuzz/common/util/IRandom.java:39]: Get rid of this extra line.

File: common/src/main/java/com/graphicsfuzz/common/util/IRandom.java [hunk 2]
```diff
@@ -24,12 +24,19 @@
 
   int nextInt(int bound);
 
+  default int nextInt(int origin, int bound) {
```
Comment (afd) [inline common/src/main/java/com/graphicsfuzz/common/util/IRandom.java:27]: Please add a JavaDoc comment indicating the range of integer that could be returned - i.e., making it clear that bound should be larger than origin, and that an integer in the range [origin, origin + bound) is returned.

File: common/src/main/java/com/graphicsfuzz/common/util/IRandom.java [hunk 3]
```diff
@@ -24,12 +24,19 @@
...
+    return (int) (nextLong(((long)bound - (long) origin)) + (long) origin);
```
Comment (afd) [inline common/src/main/java/com/graphicsfuzz/common/util/IRandom.java:28]: Precede the return by an assertion that bound > origin?

File: common/src/main/java/com/graphicsfuzz/common/util/SameValueRandom.java [hunk 4]
```diff
@@ -20,10 +20,16 @@
...
-  public SameValueRandom(boolean boolValue, int intValue) {
+  public SameValueRandom(boolean boolValue, int intValue, long longValue) {
     this.boolValue = boolValue;
     this.intValue = intValue;
+    this.longValue = longValue;
+  }
+
+  public SameValueRandom(boolean boolValue, int intValue) {
+    this(boolValue, intValue, 0L);
   }
```
Comment (afd) [inline common/src/main/java/com/graphicsfuzz/common/util/SameValueRandom.java:33]: Please get rid of this, and instead adapt each existing SameValueRandom constructor to take 0L for its third argument.

File: generator/src/test/java/com/graphicsfuzz/generator/semanticschanging/Expr2BinaryMutationFinderTest.java [hunk 5]
```diff
@@ -80,11 +80,26 @@ public int nextInt(int bound) {
         return 2;
       }
 
+      @Override
+      public int nextInt(int origin, int bound) {
+        return IRandom.super.nextInt(origin, bound);
```
Comment (afd) [inline generator/src/test/java/com/graphicsfuzz/generator/semanticschanging/Expr2BinaryMutationFinderTest.java:85]: Presumably this is never called as the original test didn't depend on it. So I suggest replacing the body with:

```
throw new RuntimeException("This method was added to the IRandom interface after this test was written and thus should not be called; if it becomes required the IRandom implementation in the test should be revised.");
```

File: generator/src/test/java/com/graphicsfuzz/generator/semanticschanging/Expr2BinaryMutationFinderTest.java [hunk 6]
```diff
@@ -80,11 +80,26 @@ public int nextInt(int bound) {
...
+      }
+
+      @Override
+      public long nextLong(long bound) {
+        return 2;
+      }
```
Comment (afd) [inline generator/src/test/java/com/graphicsfuzz/generator/semanticschanging/Expr2BinaryMutationFinderTest.java:91]: Same.

File: generator/src/test/java/com/graphicsfuzz/generator/semanticschanging/Expr2BinaryMutationFinderTest.java [hunk 7]
```diff
@@ -80,11 +80,26 @@ public int nextInt(int bound) {
...
       @Override
       public Float nextFloat() {
         throw new UnsupportedOperationException();
       }
 
+      @Override
+      public int nextPositiveInt(int bound) {
+        return IRandom.super.nextPositiveInt(bound);
+      }
+
```
Comment (afd) [inline generator/src/test/java/com/graphicsfuzz/generator/semanticschanging/Expr2BinaryMutationFinderTest.java:102]: Hmm, I might be missing something but I'm having trouble understanding why this wasn't required before your changes but is now.

File: common/src/main/java/com/graphicsfuzz/common/util/CannedRandom.java [hunk 8]
```diff
@@ -30,12 +30,12 @@
    * Constructs a canned random instance that will iterate over the given lists of integers
    * and booleans.
    *
-   * @param intsAndBools Ordered sequence of integers and booleans
+   * @param intsAndBools Ordered sequence of integers, long and booleans
```
Comment (afd) [inline common/src/main/java/com/graphicsfuzz/common/util/CannedRandom.java:33]: Change the name to `intsLongsAndBools`

File: common/src/main/java/com/graphicsfuzz/common/util/CannedRandom.java [hunk 9]
```diff
@@ -30,12 +30,12 @@
...
    */
   public CannedRandom(Object... intsAndBools) {
```
Comment (afd) [inline common/src/main/java/com/graphicsfuzz/common/util/CannedRandom.java:35]: Change the name of the parameter.

**Review by afd — CHANGES_REQUESTED** [review]
Looks good! A few changes requested.

**Comment by AaronGhost** [comment]
Done

**Review by afd — APPROVED** [review]
LGTM.

**Pull request merged**